ts/session/apis/open_group_api/sogsv3/knownBlindedkeys.ts

```diff
@@ -0,0 +1,413 @@
+
+  if (realSessionIdFound) {
+    await addCachedBlindedKey({
+      blindedId,
+      realSessionId: realSessionIdFound,
+      serverPublicKey: serverPubKey,
+    });
+    return realSessionIdFound;
+  }
+  return undefined;
+}
+
+/**
+ * When we sent a message to a sogs with blinded enable, we need to store the message with the sender being our blinded pubkey.
+ * We store that mapping <ourKey, serverPk, blindedKey> in the same cache, so we can map our own messages synced easily.
+ * This function just find if there is such a mapping already cached, but won't try to update the cache to find one.
+ */
+export function findCachedBlindedIdFromUnblinded(
+  unblindedId: string,
+  serverPubKey: string
+): string | undefined {
+  if (PubKey.hasBlindedPrefix(unblindedId)) {
+    throw new Error('findCachedBlindedIdFromUnblinded needs an unblindedID');
+  }
+  const found = assertLoaded().find(
+    m => m.serverPublicKey === serverPubKey && m.realSessionId === unblindedId
+  );
+  return found?.blindedId || undefined;
+}
+
+/**
+ * This function can be used to generate our blindedId for a sogs requiring it, and cache it.
+ */
+export async function findCachedOurBlindedPubkeyOrLookItUp(
+  serverPubKey: string,
+  sodium: LibSodiumWrappers
+): Promise<string> {
+  const ourNakedSessionID = UserUtils.getOurPubKeyStrFromCache();
+
+  if (PubKey.hasBlindedPrefix(ourNakedSessionID)) {
+    throw new Error('findCachedBlindedIdFromUnblindedOrLookItUp needs a unblindedID');
+  }
+  let found = findCachedBlindedIdFromUnblinded(ourNakedSessionID, serverPubKey);
+
+  if (found) {
+    return found;
+  }
+  const signingKeys = await UserUtils.getUserED25519KeyPairBytes();
+
+  // just to make sure the mapping was not added during last line call
+
+  if (!signingKeys) {
+    throw new Error('addSingleOutgoingMessage: getUserED25519KeyPairBytes returned nothing');
+  }
+
+  const blindedPubkeyForThisSogs = SogsBlinding.getBlindedPubKey(
+    fromHexToArray(serverPubKey),
+    signingKeys,
+    sodium
+  );
+  found = findCachedBlindedIdFromUnblinded(ourNakedSessionID, serverPubKey);
+
+  if (found) {
+    return found;
+  }
+
+  await addCachedBlindedKey({
+    blindedId: blindedPubkeyForThisSogs,
+    serverPublicKey: serverPubKey,
+    realSessionId: ourNakedSessionID,
+  });
+  return blindedPubkeyForThisSogs;
+}
+export function getCachedNakedKeyFromBlindedNoServerPubkey(blindedId: string): string | undefined {
+  if (isNonBlindedKey(blindedId)) {
+    return blindedId;
+  }
+  const found = assertLoaded().find(m => m.blindedId === blindedId);
+  return found?.realSessionId || undefined;
+}
+
+/**
+ * Can be used when we get an unblinded message to check if this is actually a reply to any of the conversation we were having with a blinded id, on any sogs
+ * @param unblindedId the blindedId of that user
+ * @param sodium passed so we can make this function not async
+ */
+export function findCachedBlindedMatchOrItLookupAllServers(
```

Comment:
Same thing as earlier comment. LookupOnAllServers might be better

Reply:
renamed to `findCachedBlindedMatchOrLookupOnAllServers`